src/category_theory/with_term.lean

```diff
@@ -0,0 +1,322 @@
+/-
+Copyright (c) 2021 Adam Topaz. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Adam Topaz
+-/
+import category_theory.limits.shapes.terminal
+
+/-!
+
+# `with_init` and `with_term`
```

Comment:
My inclination would be to write out `initial` and `terminal` in full here for consistency with the existing API. (Or, for that matter, to shorten everything to the four letter abbreviations.)